src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.DesignTime.targets

```diff
@@ -33,6 +33,21 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ProjectCapability Include="DynamicFileNesting" />
   </ItemGroup>
 
+  <ItemGroup>
+    <!--
+      Add all cshtml files to UpToDateCheckInput - a collection of files used by FastUpToDateCheck to determine
+      if any of the the project inputs have changed.
+    -->
+    <UpToDateCheckInput Condition="'$(RazorCompileOnBuild)'=='true'" Include="@(Content->WithMetadataValue('Extension', '.cshtml'))" />
+
+    <!--
+      Add Razor output files to UpToDateCheckBuilt - a collection of files used by FastUpToDateCheck to determine
+      if any of the project's outputs have changed.
+    -->
+    <UpToDateCheckBuilt Include="@(RazorIntermediateAssembly)"
```

Comment:
The primary scenario this is useful for is when the Views.dll gets out-of-sync with cshtml files (e.g. if a user deletes this file from the output directory). It would be correct to also include the `RazorTargetFile`. Unfortunately we calculate that inside a target and no target is called as part of gathering these inputs. It's a bit of an edge scenario, and we can wait for user feedback

Reply:
I'm not sure I understand what you're saying here.

Reply:
Here's the scenario: In VS,

1. File -> New WebApp
2. Build it
3. Delete obj\Debug\netcoreapp2.1\WebApp1.Views.dll
3. Build again 

Expected:
We rebuild the Views.dll

Actual:
========== Build: 0 succeeded, 0 failed, 1 up-to-date, 0 skipped ==========

This adds the file to UpToDateCheckBuilt so this causes VS to recompile. We should afford the same treatment for `bin\Debug\netcoreapp2.1\WebApp1.Views.dll`, but calculating it's path is inside a target now, hence unavailable in the global context. It's a less than common scenario so I'm not doing changing things to solve it at this point.